kubernetes/samples/scripts/domain-lifecycle/helper.sh

```diff
@@ -0,0 +1,606 @@
+      # Update policy when server name matches current server and unsetting
+      if [[ "${serverName}" == "${localServerName}" && "${withPolicy}" == "UNSET" ]]; then
+        policy=UNSET
+      fi
+      # check if server should start based on replica count, policy and current replicas
+      shouldStart "${currentReplicas}" "${policy}" "${replicaCount}" result
+      if [ "${result}" == 'True' ]; then
+        # server should start, increment current replicas and add server to list of started servers
+        currentReplicas=$((currentReplicas+1))
+        startedServers+=(${localServerName})
+      fi
+    done
+  fi
+  startedSize=${#startedServers[@]}
+  if [ ${startedSize} -gt 0 ]; then
+    # check if current server is in the list of started servers
+    if checkStringInArray ${serverName} ${startedServers[@]}; then
+      eval $__started="true"
+      return
+    fi
+  fi
+  eval $__started="false"
+}
+
+#
+# Function to check if server should start based on policy and current replicas
+# $1 - Current number of replicas
+# $2 - Server start policy
+# $3 - Replica count
+# $4 - Return value
+#
+function shouldStart {
+  local currentReplicas=$1
+  local policy=$2
+  local replicaCount=$3 
+  local __result=$4
+
+  if [ "$policy" == "ALWAYS" ]; then
+    eval $__result=True
+  elif [ "$policy" == "NEVER" ]; then
+    eval $__result=False
+  elif [ "${currentReplicas}" -lt "${replicaCount}" ]; then
+    eval $__result=True
+  else 
+    eval $__result=False
+  fi
+}
+
+#
+# Function to create patch string for updating replica count
+# $1 - Domain resource in json format
+# $2 - Name of cluster whose replica count will be patched
+# $3 - operatation string indicating whether to increment or decrement count
+# $4 - Return value containing replica update patch string
+# $5 - Retrun value containing updated replica count
+#
+function createReplicaPatch {
+  local domainJson=$1
+  local clusterName=$2
+  local operation=$3
+  local __result=$4
+  local __replicaCount=$5
+  local maxReplicas=""
+  local errorMessage="@@ ERROR: Maximum number of servers allowed (maxReplica = ${maxReplicas}) \
+are already running. Please increase cluster size to start new servers."
+
+  replicasCmd="(.spec.clusters[] \
+    | select (.clusterName == \"${clusterName}\")).replicas"
+  maxReplicaCmd="(.status.clusters[] | select (.clusterName == \"${clusterName}\")) \
+    | .maximumReplicas"
+  replica=$(echo ${domainJson} | jq "${replicasCmd}")
+  if [[ -z "${replica}" || "${replica}" == "null" ]]; then
+    replica=$(echo ${domainJson} | jq .spec.replicas)
+  fi
+  if [ "${operation}" == "DECREMENT" ]; then
+    replica=$((replica-1))
+    if [ ${replica} -lt 0 ]; then
+      replica=0
+    fi
+  elif [ "${operation}" == "INCREMENT" ]; then
+    replica=$((replica+1))
+    maxReplicas=$(echo ${domainJson} | jq "${maxReplicaCmd}")
+    if [ ${replica} -gt ${maxReplicas} ]; then
+      echo "${errorMessage}"
```

Comment:
1) should this be a 'printError'?  Or maybe it should just be aan INFO?  See my new comment on the caller of this function.
2) maybe also set __replicaCount to some dummy value?